include/malloc_interceptor.h

```diff
@@ -0,0 +1,59 @@
+#ifndef __MALLOC_INTERCEPTOR_H
+#define __MALLOC_INTERCEPTOR_H
+#define _GNU_SOURCE
```

Comment:
Defining `_GNU_SOURCE` inside a header file is generally discouraged because it affects the compilation of any file that includes it. We should move it to the .c file.